frontend/src/views/Management/Company/CompanyForm.tsx

```diff
@@ -0,0 +1,127 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { SaveSharp, Undo } from '@mui/icons-material';
+import { Box, Button, CircularProgress, TextField, css } from '@mui/material';
+import { useEffect } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router';
+
+import { trpc } from '@frontend/client';
+import { FormField } from '@frontend/components/forms';
+import { SectionTitle } from '@frontend/components/forms/SectionTitle';
+import { useNotifications } from '@frontend/services/notification';
+import { useTranslations } from '@frontend/stores/lang';
+import { getRequiredFields } from '@frontend/utils/form';
+
+import { Company, companySchema } from '@shared/schema/contractor';
+
+interface Props {
+  businessId?: string;
+  onSubmitted?: () => void;
+}
+
+export function CompanyForm(props: Props) {
+  const tr = useTranslations();
+  const notify = useNotifications();
+  const navigate = useNavigate();
+
+  const company = trpc.contractor.getCompanyById.useQuery(
+    { businessId: props.businessId } as { businessId: string },
+    { enabled: !!props.businessId }
+  );
+
+  useEffect(() => {
+    if (props.businessId && company.data) {
+      form.reset(company.data);
+    }
+  }, [company.data, props.businessId]);
+
+  const form = useForm<Company>({
+    mode: 'all',
+    resolver: zodResolver(companySchema),
+    context: {
+      requiredFields: getRequiredFields(companySchema),
+    },
+    defaultValues: company.data ?? {
+      companyName: '',
+      businessId: '',
+    },
+  });
+
+  const companyUpsert = trpc.contractor.upsertCompany.useMutation({
+    onSuccess: (data) => {
```

Comment:
[nit] Seems like the type of `data` is `void` and it's unused, so it can be removed from here.
```suggestion
    onSuccess: () => {
```